components/camel-consul/src/main/java/org/apache/camel/component/consul/endpoint/ConsulEventConsumer.java

```diff
@@ -98,8 +99,8 @@ private void onEvent(Event event) {
             if (event.getTagFilter().isPresent()) {
                 message.setHeader(ConsulConstants.CONSUL_TAG_FILTER, event.getTagFilter().get());
             }
-
-            message.setBody(event.getPayload().orNull());
+            
+            message.setBody(Optional.ofNullable(event.getPayload()));
 
```

Comment:
Doesn't looks similar, if I'm not mistaking this

    event.getPayload().orNull()

returns the event payload or null, whereas

    message.setBody(Optional.ofNullable(event.getPayload()));

creates an eventually empty Optional